guardrails/llm_providers.py

```diff
@@ -364,6 +484,31 @@ def get_llm_ask(llm_api: Callable, *args, **kwargs) -> PromptCallableBase:
     except ImportError:
         pass
 
+    try:
+        from transformers import (  # noqa: F401 # type: ignore
+            FlaxPreTrainedModel,
+            PreTrainedModel,
+            TFPreTrainedModel,
+        )
+
+        api_self = getattr(llm_api, "__self__", None)
+
+        if (
+            isinstance(api_self, PreTrainedModel)
+            or isinstance(api_self, TFPreTrainedModel)
+            or isinstance(api_self, FlaxPreTrainedModel)
+        ):
+            return HuggingFaceModelCallable(*args, model_generate=llm_api, **kwargs)
```

Comment:
Similar thing as the other comment: Before calling the `HuggingFaceModelCallable`, as we're already checking whether the model type is correct, can we also check whether `llm_api` e.g `model.generate` is a valid call for that model? Else an unknown error might get raised when actually calling that method above on line 353 in `HuggingFaceModelCallable`'s `invoke_llm`.